Flagsmith.FlagsmithClient/FlagsmithClient.cs

```diff
@@ -37,35 +36,75 @@ namespace Flagsmith
-        private Func<string, IFlag>? DefaultFlagHandler { get; set; }
-        private ILogger? Logger { get; set; }
-        private bool EnableAnalytics { get; set; }
-        private double? RequestTimeout { get; set; }
-        private int Retries { get; set; }
-        private CacheConfig CacheConfig { get; set; }
-        private Dictionary<string, string>? CustomHeaders { get; set; }
         private EnvironmentModel? Environment { get; set; }
         private Dictionary<string, IdentityModel>? IdentitiesWithOverridesByIdentifier { get; set; }
-        private bool OfflineMode { get; set; }
-        const string DefaultApiUrl = "https://edge.api.flagsmith.com/api/v1/";
 
-        private readonly HttpClient _httpClient;
-        private readonly PollingManager? _pollingManager;
-        private readonly IEngine _engine;
-        private readonly AnalyticsProcessor? _analyticsProcessor;
-        private readonly RegularFlagListCache? _regularFlagListCache; private readonly ConcurrentDictionary<string, IdentityFlagListCache>? _flagListCacheDictionary;
-        private readonly BaseOfflineHandler? _offlineHandler;
+        private readonly FlagsmithConfiguration _config;
+        private readonly IEngine _engine = new Engine();
+        private PollingManager? _pollingManager;
+        private AnalyticsProcessor? _analyticsProcessor;
+        private RegularFlagListCache? _regularFlagListCache;
+        private ConcurrentDictionary<string, IdentityFlagListCache>? _flagListCacheDictionary;
+
+        private void Initialise()
+        {
+            if (_config.OfflineMode && _config.OfflineHandler is null)
+            {
+                throw new Exception("ValueError: offlineHandler must be provided to use offline mode.");
+            }
+            else if (_config.DefaultFlagHandler != null && _config.OfflineHandler != null)
+            {
+                throw new Exception("ValueError: Cannot use both defaultFlagHandler and offlineHandler.");
+            }
+
+            if (_config.OfflineHandler != null)
+            {
+                Environment = _config.OfflineHandler.GetEnvironment();
+            }
+
+            if (!_config.OfflineMode)
+            {
+                if (string.IsNullOrEmpty(_config.EnvironmentKey))
+                {
+                    throw new Exception("ValueError: environmentKey is required");
+                }
+                if (_config.EnableAnalytics)
+                    _analyticsProcessor = new AnalyticsProcessor(_config.HttpClient, _config.EnvironmentKey, _config.ApiUri.ToString(), _config.Logger, _config.CustomHeaders);
+
+                if (_config.EnableLocalEvaluation)
+                {
+                    if (!_config.EnvironmentKey!.StartsWith("ser."))
+                    {
+                        Console.WriteLine(
+                            "In order to use local evaluation, please generate a server key in the environment settings page."
+                        );
```

Comment:
This may conflict with #141, but as per the process laid out in this PR to _revert_ that change, I think we should keep this Console.WriteLine until we are ready to remove the deprecated methods in this PR.